### PR TITLE
API Get developer details

### PR DESCRIPTION
**Pull request by thupv**
# Purpose of PR
Search function

### CODE REVIEW

File: app/controllers/api/developer_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          render json: DeveloperSerializer.new(@developer).serialized_json
+        end
+      end
+      # rubocop:enable Style/GuardClause
+    end
+
+    private
+
+    def set_developer
+      @developer = Developer.includes(:developer_languages, :developer_programmings).find(params[:id])
```
Comment (thienhv) [inline app/controllers/api/developer_controller.rb]: @thupv 

is `includes(:developer_languages, :developer_programmings)` necessary ?

File: spec/requests/api/developer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            }
+          }
+        }
+      end
+      it_behaves_like 'http_status_code_200_with_json'
+    end
+
+    context 'when the post does not exist' do
+      before do
+        get '/api/developer/non_exist_id'
```
Comment (jp-ryuji) [inline spec/requests/api/developer_spec.rb]: Should be `non_existing_id`.
`exist` is a verb.

File: spec/requests/api/developer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+RSpec.describe 'Api::DeveloperController', type: :request do
+  include_context 'api'
+
+  describe 'GET /api/developer/:developer_id' do
+    before do
+      @language = create(:language, code: 'vn')
+      @programming_language = create(:programming_language, name: 'ruby')
+      @developer = create(:developer,
+                          email: 'abc2321@gmail.com',
```
Comment (jp-ryuji) [inline spec/requests/api/developer_spec.rb]: This is a small thing.
But I recommend you to use more meaningless value for email, `abc2321@gmail.com`.
Such as `example`, `test`, `foo` or `bar`, like `foo@example.com`.
Then other developers understand the value doesn't mean anything.

Moreover, you don't need to set email here.
It should be set by factory_bot if you define it.
Can you try this approach?

File: spec/controllers/developer_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (jp-ryuji) [inline spec/controllers/developer_controller_spec.rb]: You should remove unused files.

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
+  namespace :api do
```
Comment (jp-ryuji) [inline config/routes.rb:2]: (REST) api should have versions.

File: spec/helpers/developer_helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
```
Comment (jp-ryuji) [inline spec/helpers/developer_helper_spec.rb]: You should remove unused files.

File: app/helpers/developer_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module DeveloperHelper
```
Comment (jp-ryuji) [inline app/helpers/developer_helper.rb]: You should remove unused files.

File: spec/requests/api/developer_spec.rb [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+            }]
+        }
+      end
+      let(:expected_response) do
+        {
+          data: {
+            id: @developer.id.to_s,
+            type: 'developer',
+            attributes: {
+              email: @developer.email
```
Comment (jp-ryuji) [inline spec/requests/api/developer_spec.rb:44]: The details should be returned.
Not only email.

File: spec/requests/api/developer_spec.rb [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+              languages: language,
+              programming_languages: programming_language
+            }
+          }
+        }
+      end
+      it_behaves_like 'http_status_code_200_with_json'
+    end
+
+    context 'when the post does not exist' do
```
Comment (jp-ryuji) [inline spec/requests/api/developer_spec.rb]: post?

**Review by jp-ryuji — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 10]
```diff
@@ -30,6 +30,7 @@ gem 'jbuilder', '~> 2.5'
 gem 'dotenv-rails'
 gem 'virtus'
 
+gem 'fast_jsonapi'
```
Comment (jp-ryuji) [inline Gemfile]: @thupv 
Good try with `fast_jsonapi`. 
I like the attitude. 👍

File: spec/requests/api/developer_spec.rb (hunk 8 above)
Comment (jp-ryuji) [reply spec/requests/api/developer_spec.rb:44]: Sorry.
Forget about the comment.
There's only `email` column.

File: config/routes.rb (hunk 5 above)
Comment (thupv) [reply config/routes.rb:2]: I added version but github don't recognise, could you check here: https://github.com/thupv/FindDeveloper/pull/1/files

**Review by thienhv — APPROVED** [review]

**Review by jp-ryuji — APPROVED** [review]

**Pull request merged**